src/world_interface/world_interface.h

```diff
@@ -35,6 +36,9 @@ extern const WorldInterface WORLD_INTERFACE;
 const DiffDriveKinematics& driveKinematics();
 const DiffWristKinematics& wristKinematics();
 
+// A mapping of (motor_id, shared pointer to object of the motor)
+std::unordered_map<robot::types::motorid_t, std::shared_ptr<robot::base_motor>> motor_ptrs;
```

Comment:
This shouldn't be defined in a header file, it should be private to each world interface implementation. Just remove this line and copy it to the top of `real_world_interface.cpp` and `simulator_interface.cpp`